extensions/positron-assistant/src/test/anthropic.test.ts

```diff
@@ -0,0 +1,146 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2025 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as positron from 'positron';
+import * as vscode from 'vscode';
+import * as sinon from 'sinon';
+import { AnthropicLanguageModel } from '../anthropic';
+import { ModelConfig } from '../config';
+
+suite('AnthropicLanguageModel', () => {
+	let model: AnthropicLanguageModel;
+	let mockClient: any;
+	let mockProgress: vscode.Progress<vscode.ChatResponseFragment2>;
+	let mockCancellationToken: vscode.CancellationToken;
+
+	setup(() => {
+		// Create a mock Anthropic client
+		mockClient = {
+			messages: {
+				stream: sinon.stub().returns({
+					on: sinon.stub(),
+					abort: sinon.stub(),
+					done: sinon.stub().resolves()
+				})
+			}
+		};
+
+		// Create a mock configuration
+		const config: ModelConfig = {
+			id: 'test-model',
+			name: 'Test Model',
+			provider: 'anthropic',
+			model: 'claude-test',
+			apiKey: 'test-api-key', // pragma: allowlist secret
+			type: positron.PositronLanguageModelType.Chat
+		};
+
+		// Create an instance of the AnthropicLanguageModel
+		model = new AnthropicLanguageModel(config);
+
+		// Replace the client with our mock
+		(model as any)._client = mockClient;
+
+		// Create mock progress and cancellation token
+		mockProgress = {
+			report: sinon.stub()
+		};
+
+		mockCancellationToken = {
+			isCancellationRequested: false,
+			onCancellationRequested: sinon.stub()
+		};
```

Comment:
Nit: we could probably use a real cancellation token here, since we can control when it's fired via the token source:

```typescript
const cancellationTokenSource = new vscode.CancellationTokenSource();
token = cancellationTokenSource.token;
```